allauth/utils.py

```diff
@@ -264,7 +264,7 @@ def build_absolute_uri(request, location, protocol=None):
     from .account import app_settings as account_settings
 
     if request is None:
-        site = Site.objects.get_current()
+        site = get_current_site(request)
```

Comment:
`None` is not an acceptable input to `get_current_site`

Reply:
It is if `django.contrib.sites` is installed. If you don't have it installed, you can't use `build_absolute_uri` outside a request context, which makes perfect sense (where would it get domain information from?).

Reply:
I understand,  but we need to be certain that allauth is properly handing over a request everywhere. Also, this method is documented as:

    Like request.build_absolute_uri, but gracefully handling
    the case where request is None.

With these changes, that is no longer true. So the docstring needs updating...

Reply:
I'm not sure what you mean, to be honest.

The only place where this is called with `request=None` is from `get_email_confirmation_url` if you're doing email confirmation emails outside a request context. So in this case, we'll have to construct a URL from the current Django site. If you don't have one set up, or don't have `django.contrib.sites` installed, it won't work. This also didn't work before my patch.

The only thing I see that we could maybe explain a bit better is that you can't do outside-request-context email confirmations without `django.contrib.sites` installed. Is this what you want me to do? :)